byterun/win32.c

```diff
@@ -914,3 +919,64 @@ void caml_restore_win32_terminal(void)
   if (startup_codepage != 0)
     SetConsoleOutputCP(startup_codepage);
 }
+
+/* Detect if a named pipe corresponds to a Cygwin/MSYS pty: see
+
+   https://github.com/mirror/newlib-cygwin/blob/00e9bf2/winsup/cygwin/dtable.cc#L932
+*/
+typedef
+BOOL (WINAPI *tGetFileInformationByHandleEx)(HANDLE, FILE_INFO_BY_HANDLE_CLASS,
+                                             LPVOID, DWORD);
+
+static int caml_win32_is_cygwin_pty(HANDLE hFile)
+{
+  char buffer[1024];
+  FILE_NAME_INFO * nameinfo = (FILE_NAME_INFO *) buffer;
+  static tGetFileInformationByHandleEx pGetFileInformationByHandleEx = NULL;
+
+  if (pGetFileInformationByHandleEx == NULL)
```

Comment:
... and compare that here instead of `NULL` (means that the lookup only happens once, even on Windows XP)... but it's up to you if you want to use keypresses making Windows XP execution "faster" :wink:

Reply:
Sure, done.